optaplanner-examples/src/main/java/org/optaplanner/examples/cheaptime/domain/CheapTimeSolution.java

```diff
@@ -154,6 +154,7 @@ public void setScore(HardMediumSoftLongScore score) {
         facts.addAll(taskList);
         facts.addAll(taskRequirementList);
         facts.addAll(periodPowerPriceList);
+        facts.add(this);
```

Comment:
I am personally not a fan of adding the solution instance in the problem facts, but there's nothing wrong with that: OptaPlanner should support it completely.
Merging anyway, of course.